include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -3370,9 +3370,11 @@ FEEvaluationBase<dim, n_components_, Number, is_face, VectorizedArrayType>::
                                            *this->dof_info);
     }
 
+  const bool is_neighbor_cells = !is_face && !this->is_interior_face();
```

Comment:
I know this is pre-existing due to #14119, but I must say that the name is really confusing. How about using `use_unsorted_cells`? (I am not super happy to have an additional statement in the critical path also for the contiguous storage case. I think we need to work on reducing overhead in the near future.)

Reply:
You could also add an additional code comment. I had to look up in many places why we combine `!is_face` with information on a face, `is_interior_face()`.

Reply:
I think we could avoid this by setting `this->cell = numbers::invalid_unsigned_int` in `FEEvaluationShift` https://github.com/dealii/dealii/blob/22a5b07781325d0a5498de5a5e60fecd891a6ff1/tests/matrix_free/fe_evaluation_shift.cc#L188 which should choose the same code path as https://github.com/dealii/dealii/blob/22a5b07781325d0a5498de5a5e60fecd891a6ff1/include/deal.II/matrix_free/fe_evaluation.h#L2145
Would you prefer this solution?

Reply:
Yes, I think that would be better, as the information does not make sense anyway.